packages/common/src/hardhat/tasks/compile.js

```diff
@@ -32,17 +32,17 @@ internalTask(TASK_COMPILE_SOLIDITY_GET_SOURCE_PATHS, async (_, { config, network
       }
       return false;
     });
+    return filePaths;
+  } else {
```

Comment:
@chrismaree I had to make this change because `BridgePool` was getting compiled for the `--network optimism` flag but it uses Solidity 0.8. I think this change is ok because any contracts that are supposed to be compiled for the OVM should be in the `compileWhitelist`. If there is no `compileWhitelist`, then compile ALL contracts plus these extra `contracts-ovm` ones.